lib/core.js

```diff
@@ -1745,6 +1745,11 @@ type IteratorResult<+Yield,+Return> =
 interface $Iterator<+Yield,+Return,-Next> extends $Iterable<Yield,Return,Next> {
     @@iterator(): $Iterator<Yield,Return,Next>;
     next(value?: Next): IteratorResult<Yield,Return>;
+    /**
+     * Calls a defined callback function on each element returned from this iterator.
+     * @param callbackfn A function that accepts up to two arguments: the yielded value, and it's index. The map method calls the callbackfn function once per iteration.
+     */
+    map<U>(callbackfn: (value: Yield, index: number) => U): $Iterator<U, void, mixed>;
```

Comment:
The `Next` type of the returned iterator is funny.

According to [the spec discussion](https://github.com/tc39/proposal-iterator-helpers/blob/main/DETAILS.md#passing-the-protocol) it should match the type of source iterator. In practice, Chrome discards any values passed to the `next` method of the returned iterator.

[TypeScript currently uses `mixed`](https://github.com/microsoft/TypeScript/blob/d85767abfd83880cea17cea70f9913e9c4496dcc/src/lib/esnext.iterator.d.ts#L32), which existing behavior tolerates. It also tolerates working code if implementations eventually match the spec, albeit with reduced type safety.

We do the same here.